.github/workflows/deploy-cloud.yaml

```diff
@@ -34,32 +40,51 @@ jobs:
       - name: Build
         run: |
           npm run build
+
+      - name: Configure AWS Credentials
+        if: env.REGION == 'eu'
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-access-key-id: ${{ secrets.AWS_CLOUD_EU_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_CLOUD_EU_SECRET_ACCESS_KEY }}
+          aws-region: eu-west-1
+
       - name: Configure AWS Credentials
-        uses: aws-actions/configure-aws-credentials@v1
+        if: env.REGION == 'us'
+        uses: aws-actions/configure-aws-credentials@v4
         with:
-          aws-access-key-id: ${{ secrets.AWS_DASHBOARD_PROD_ACCESS_KEY_ID }}
-          aws-secret-access-key: ${{ secrets.AWS_DASHBOARD_PROD_SECRET_ACCESS_KEY }}
+          aws-access-key-id: ${{ secrets.AWS_CLOUD_US_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_CLOUD_US_SECRET_ACCESS_KEY }}
           aws-region: us-east-1
+
       - name: Deploy
+        if: env.REGION == 'eu'
         run: |
-          if [[ "$REGION" == "us" ]]; then
-            BUCKET=${{ secrets.AWS_PROD_US_DEPLOYMENT_BUCKET }}
-            CF_ID=${{ secrets.AWS_PROD_US_CF_DIST_ID }}
-          elif [[ "$REGION" == "eu" ]]; then
-            BUCKET=${{ secrets.AWS_PROD_EU_DEPLOYMENT_BUCKET }}
-            CF_ID=${{ secrets.AWS_PROD_EU_CF_DIST_ID }}
-            CF_2_ID=${{ secrets.AWS_PROD_EU_CF_2_DIST_ID }}
-          else
-            echo "Unknown region provided"
-            exit 1
-          fi
+          aws s3 sync build/dashboard "s3://${DEPLOYMENT_BUCKET}/${ENVIRONMENT}/static/"
+          aws s3 cp build/dashboard/index.html "s3://${DEPLOYMENT_BUCKET}/${ENVIRONMENT}/"
+
+          IFS=',' read -ra DISTRIBUTIONS_ARRAY <<< "${CDN_DISTRIBUTIONS}"
+          for DISTRIBUTION in "${DISTRIBUTIONS_ARRAY[@]}"; do
+            for i in {1..3}; do aws cloudfront create-invalidation --distribution-id "$DISTRIBUTION" --paths "/dashboard*" && break || sleep 5; done
+          done
+          env:
+            DEPLOYMENT_BUCKET: ${{ secrets.AWS_CLOUD_EU_DEPLOYMENT_BUCKET }}
+            CDN_DISTRIBUTIONS: ${{ secrets.AWS_CLOUD_EU_CDN_DISTRIBUTIONS }}
+
+            - name: Deploy
+            if: env.REGION == 'us'
+            run: |
+              aws s3 sync build/dashboard "s3://${DEPLOYMENT_BUCKET}/${ENVIRONMENT}/static/"
+              aws s3 cp build/dashboard/index.html "s3://${DEPLOYMENT_BUCKET}/${ENVIRONMENT}/"
+
+              IFS=',' read -ra DISTRIBUTIONS_ARRAY <<< "${CDN_DISTRIBUTIONS}"
+              for DISTRIBUTION in "${DISTRIBUTIONS_ARRAY[@]}"; do
+                for i in {1..3}; do aws cloudfront create-invalidation --distribution-id "$DISTRIBUTION" --paths "/dashboard*" && break || sleep 5; done
```

Comment:
Instead of duplicating code across all branches and workflows, you could use composite actions

Reply:
I agree with that and if that would be only-SRE-maintaned project I would try to do that, as that's something that @saleor/dashboard team owns, I would prefer to leave that for them to address this duplication.

Reply:
This will also require consideration about main staging/fake production (where we deploy to two S3 targets, but invalidate cache only on one distribution.